### PR TITLE
Add PsaSignMessage and PsaVerifyMessage operations

### PR DESCRIPTION
**Pull request by akazimierskigl**
Implementation of PsaSignMessage and PsaVerifyMessage operations.
It requires updated rust-psa-crypto with new functions to compile.

### CODE REVIEW

**Comment by hug-dev** [comment]
When https://github.com/parallaxsecond/rust-psa-crypto/pull/83 is merged, feel free to import the changes in the `Cargo.toml` here by doing something like in [here](https://github.com/parallaxsecond/parsec-interface-rs/commit/b6740e4c0896553c1f0650eaad96ecc4f8313c85#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542). Then we can run the CI again to make sure it passes.

**Comment by hug-dev** [comment]
I am adressing the clippy failures in #106, you will be able to rebase on that after it's merged :)

File: src/operations/psa_sign_message.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+/// Native object for asymmetric sign operations.
+#[derive(Debug)]
+pub struct Operation {
+    /// Defines which key should be used for the signing operation.
+    pub key_name: String,
+    /// An asymmetric signature algorithm that separates the hash and sign operations, that is
+    /// compatible with the type of key.
+    pub alg: AsymmetricSignature,
+    /// The input whose signature is to be verified. This is usually the hash of a message.
```
Comment (hug-dev) [inline src/operations/psa_sign_message.rs:18]: I think this should be something like:
```
/// The message to sign
```

File: src/operations/psa_verify_message.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Contributors to the Parsec project.
```
Comment (hug-dev) [inline src/operations/psa_verify_message.rs:1]: Copyright year here should be 2021 too (same for all new files)

File: src/operations/psa_verify_message.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//! # PsaVerifyMessage operation
+//!
+//! Verify the signature of a hash or short message using a public key.
```
Comment (hug-dev) [inline src/operations/psa_verify_message.rs:5]: Should be:
```
Verify the signature of a message using a public key.
```

File: src/operations/psa_verify_message.rs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+/// Native object for asymmetric verification of signatures.
+#[derive(Debug)]
+pub struct Operation {
+    /// `key_name` specifies the key to be used for verification.
+    pub key_name: String,
+    /// An asymmetric signature algorithm that separates the hash and sign operations, that is
+    /// compatible with the type of key.
+    pub alg: AsymmetricSignature,
+    /// The `message` contains a short message for the
+    /// asymmetric signing operation.
```
Comment (hug-dev) [inline src/operations/psa_verify_message.rs:18]:
```
/// The message whose signature is to be verified
```

**Comment by hug-dev** [comment]
Ah yes in the `convert_...` files, do you mind adding simple tests (one or two per file) for the conversions between protobuf <-> Parsec structures? Most of it can be copied from sign/verify_hash I would hope.

**Review by hug-dev — APPROVED** [review]
Great, thanks a lot for the effort of adding all the tests 👌

**Review by ionut-arm — APPROVED** [review]
👍🏻

**Pull request merged**